mesop/dataclass_utils/dataclass_utils.py

```diff
@@ -103,6 +103,13 @@ def _recursive_update_dataclass_from_json_obj(instance: Any, json_dict: Any):
       else:
         # For other types, set the value directly.
         setattr(instance, key, value)
+    else:
+      if isinstance(instance, dict):
```

Comment:
Comment for line 92-93 -> Is the comment there still relevant?

Reply:
good question - i've updated the comment to clarify the assumption.